### ISSUE TITLE
RUF006: False positive warning for task stored in global variable

### ISSUE THREAD

**Issue by 5j9** [astral-sh/ruff#18012]
### Summary

Consider the following sample code:
```python
from asyncio import create_task, run, sleep

task = create_task(sleep(2))
run(sleep(2))
```

Despite `task` being a global variable, Ruff reports the following error:

```bash
$ ruff check --isolated \~scratch.py --select RUF006
~scratch.py:3:14: RUF006 Store a reference to the return value of `asyncio.create_task`
  |
1 | from asyncio import create_task, run, sleep
2 |
3 | watch_task = create_task(sleep(2))
  |              ^^^^^^^^^^^^^^^^^^^^^ RUF006
4 | run(sleep(2))
  |

Found 1 error.
```
**Expected Behavior**
Since `task` is global, storing a reference in it should not be problematic. Ruff should not flag this code.

### Version

ruff 0.11.9

**Comment by MichaReiser** [comment]
I think that change would be okay. Although this might also be a situation where it's fine to use a `noqa`.
